### PR TITLE
Typescript Vulnerability Fixed

### PR DESCRIPTION
**Pull request by sadiqboxed**
We want to upgrade to Redis 4.0.2 since there is a security vulnerability in 3.0.2 (on the boxed-services side). However there has been some library changes with that package that breaks the usage of warlock in v4. There isnt any code in the open sourced version of the repo that is meant to rectify it so we needed to update the code here. 

There was also a security vulnerability with the lua script loading/running library in this version of the package. We removed that package and replaced it with a safe code. 

We keep backwards compatibility with anyone currently calling warlock with an untyped/old redis instance as well (while fixing the lua script issue). If you can find a version of redis under 4 that doesnt have a vulnerability then you should be able to plug that version into warlock and have it work.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,13 +29,18 @@
...
-    "should": "^3.1.3",
-    "mocha": "^1.17.1",
-    "redis": "^0.10.1",
-    "async": "^0.9.0"
+    "@types/mocha": "7.0.0",
+    "@types/redis": "2.8.0",
+    "async": "3.1.0",
+    "chai": "4.3.4",
+    "mocha": "9.1.4",
+    "redis": "2.8.0",
```
Comment (sadiqboxed) [inline package.json:41]: Not sure if CI will complain about security vulns in devDependencies. (If it gets packaged in dist for some reason then it should) If it does then we might have to get rid of this (either keep everything untyped or remove backwards compat)

File: src/WarlockError.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
```
Comment (sadiqboxed) [inline src/WarlockError.ts:1]: From previous branch

File: src/warlock.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
+import util, { callbackify, promisify } from 'util';
```
Comment (sadiqboxed) [inline src/warlock.ts:1]: From previous branch.

**Review by chrissung — APPROVED** [review]
👏 🔥  so does this become v1.1 or v2.0?

**Comment by sadiqboxed** [comment]
1.1? I figured we can go to 2 when we have breaking changes.

**Pull request merged**